src/Adapter/Presenter/Cart/CartPresenter.php

```diff
@@ -570,19 +570,6 @@ private function getTemplateVarVouchers(Cart $cart)
             } else {
                 $freeShippingOnly = false;
                 $totalCartVoucherReduction = $this->includeTaxes() ? $cartVoucher['value_real'] : $cartVoucher['value_tax_exc'];
-                $currencyFrom = new \Currency($cartVoucher['reduction_currency']);
```

Comment:
I'm not sure you should remove it all 🤔  But maybe rather check if the currencies ids match, if they don't then you need to perform the conversion You can also use `Tools::convertPriceFull` which (I believe) manages all these checks and conversion

Reply:
I've tried many combinations with different currencies, different discount currencies, different kind of discounts and each time it worked well without performing any other conversion.

Reply:
So does it mean `$cart->getCartRules()` already performs the conversion?
If this is something we changed we might need to mark this as a BC (if developers expect it to return not converted values)

Reply:
Seems like it's a recent change (introduced in 177) by this PR https://github.com/PrestaShop/PrestaShop/pull/14263 So I guess we just need to ensure that it doesn't create a regression related to this issue https://github.com/PrestaShop/PrestaShop/issues/9914

Reply:
I'll add it in the *how to test* section 😉